packages/venia-concept/src/components/Checkout/Receipt/receipt.js

```diff
@@ -1,93 +1,91 @@
 import Button from 'src/components/Button';
 import defaultClasses from './receipt.css';
 
-export const CONTINUE_SHOPPING_BUTTON_ID = 'continue-shopping-button';
+export const VIEW_ORDER_DETAILS_BUTTON_ID = 'view-order-details-button';
 export const CREATE_ACCOUNT_BUTTON_ID = 'create-account-button';
 
-class Receipt extends Component {
-    static propTypes = {
-        classes: shape({
-            body: string,
-            footer: string,
-            root: string
-        }),
-        continueShopping: func.isRequired,
-        order: shape({
-            id: string
-        }).isRequired,
-        createAccount: func.isRequired,
-        reset: func.isRequired,
-        user: shape({
-            isSignedIn: bool
-        })
-    };
+const Receipt = props => {
+    const { createAccount, history, reset, user } = props;
 
-    static defaultProps = {
-        order: {},
-        continueShopping: () => {},
-        reset: () => {},
-        createAccount: () => {}
-    };
+    const classes = mergeClasses(defaultClasses, props.classes);
 
-    componentWillUnmount() {
-        this.props.reset();
-    }
+    useEffect(() => {
+        return () => {
+            reset();
+        };
+    }, [reset]);
 
-    createAccount = () => {
-        this.props.createAccount(this.props.history);
-    };
+    const handleCreateAccount = useCallback(() => {
+        createAccount(history);
+    }, [createAccount, history]);
 
-    continueShopping = () => {
-        this.props.continueShopping(this.props.history);
-    };
+    const handleViewOrderDetails = useCallback(() => {
+        // TODO: Implement/connect/redirect to order details page.
+    }, [props.order]);
 
-    render() {
-        const {
-            classes,
-            order: { id },
-            user
-        } = this.props;
-
-        return (
-            <div className={classes.root}>
-                <div className={classes.body}>
-                    <h2 className={classes.header}>
-                        Thank you for your purchase!
-                    </h2>
-                    <div className={classes.textBlock}>
-                        Your order # is{' '}
-                        <span className={classes.orderId}>{id}</span>
-                        <br />
-                        We&rsquo;ll email you an order confirmation with details
-                        and tracking info
-                    </div>
+    return (
+        <div className={classes.root}>
+            <div className={classes.body}>
+                <h2 className={classes.header}>Thank you for your purchase!</h2>
+                <div className={classes.textBlock}>
+                    You will receive an order confirmation email with order
+                    status and other details.
+                    <br />
+                    {user.isSignedIn &&
+                        'You can also visit your account page for more information.'}
+                </div>
+                {user.isSignedIn && (
```

Comment:
Can we adjust styles to get rid of these repetitive ifs, and use a ternary?

```jsx
{user.isSignedIn ? (
    // signed in logic
) : (
    // not signed in logic
)}
```

Reply:
Sorry, my intention here was to just contain all the signed in logic, something like this.

```jsx
return (
    <div className={classes.root}>
        <div className={classes.body}>
            <h2 className={classes.header}>Thank you for your purchase!</h2>
            <div className={classes.textBlock}>
                You will receive an order confirmation email with order
                status and other details.
            </div>
            {user.isSignedIn ? (
                <Fragment>
                    <div className={classes.textBlock}>
                        You can also visit your account page for more information.
                    </div>
                    <Button
                        data-id={VIEW_ORDER_DETAILS_BUTTON_ID}
                        onClick={handleViewOrderDetails}
                    >
                        View Order Details
                    </Button>
                </Fragment>
            ) : (
                <Fragment>
                    <hr />
                    <div className={classes.textBlock}>
                        Track order status and earn rewards for your
                        purchase by creating and account.
                    </div>
                    <Button
                        data-id={CREATE_ACCOUNT_BUTTON_ID}
                        priority="high"
                        onClick={handleCreateAccount}
                    >
                        Create an Account
                    </Button>
                </Fragment>
            )}
        </div>
    </div>
);
```